hack/setup-tracing.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -euf -o pipefail
+
+# Default Values
+NAMESPACE="kube-system"
+TRACING_FILES=("observability/tracing/otel-configmap.yaml"
+               "observability/tracing/otel-deployment.yaml"
+               "observability/tracing/otel-service.yaml"
+               "observability/tracing/jager-deployment.yaml"
+               "observability/tracing/jager-service.yaml")
+
+# Ensure namespace exists
+if ! kubectl get namespace "${NAMESPACE}" > /dev/null 2>&1; then
+  echo "Namespace '${NAMESPACE}' does not exist. Creating..."
+  kubectl create namespace "${NAMESPACE}"
+else
+  echo "Namespace '${NAMESPACE}' already exists."
+fi
+
+# Apply each file
+echo "Applying tracing YAML files..."
+for file in "${TRACING_FILES[@]}"; do
+  if [[ -f "$file" ]]; then
+    echo "Applying $file..."
+    kubectl apply -f "$file" --namespace ${NAMESPACE}
+  else
+    echo "Error: File $file not found. Exiting..."
+    exit 1
+  fi
+done
+
+# Retrieve and print the Jaeger LoadBalancer IP
+get_jaeger_lb_ip() {
+  kubectl get svc jaeger-collector -n ${NAMESPACE} -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
+}
+
+echo "Waiting for Jaeger LoadBalancer to get an external IP..."
+EXTERNAL_IP=""
+while [[ -z "$EXTERNAL_IP" ]]; do
+  EXTERNAL_IP=$(get_jaeger_lb_ip)
+  if [[ -z "$EXTERNAL_IP" ]]; then
+    echo "Waiting for LoadBalancer external IP..."
+    sleep 10
+  fi
+done
```

Comment:
question: Is this going infinite loop? May be we do set amount of retries or have some backoff retry logic here?

Reply:
I've updated the loop with number of retires before backing off